src/voice/VoiceRecorder.ts

```diff
@@ -111,6 +127,37 @@ export class VoiceRecorder {
         return this.mxc;
     }
 
+    private tryUpdateLiveData = (ev: AudioProcessingEvent) => {
+        if (!this.recording) return;
+
+        // The time domain is the input to the FFT, which means we use an array of the same
+        // size. The time domain is also known as the audio waveform. We're ignoring the
+        // output of the FFT here (frequency data) because we're not interested in it.
+        //
+        // We use bytes out of the analyser because floats have weird precision problems
+        // and are slightly more difficult to work with. The bytes are easy to work with,
+        // which is why we pick them (they're also more precise, but we care less about that).
+        const data = new Uint8Array(this.recorderFFT.fftSize);
+        this.recorderFFT.getByteTimeDomainData(data);
+
+        // Because we're dealing with a uint array we need to do math a bit differently.
+        // If we just `Array.from()` the uint array, we end up with 1s and 0s, which aren't
+        // what we're after. Instead, we have to use a bit of manual looping to correctly end
+        // up with the right values
+        const translatedData: number[] = [];
+        for (let i = 0; i < data.length; i++) {
+            // All we're doing here is inverting the amplitude and putting the metric somewhere
+            // between zero and one. Without the inversion, lower values are "louder", which is
+            // not super helpful.
+            translatedData.push(1 - (data[i] / 128.0));
```

Comment:
Isn't this turning the bytes into the same thing you'd get out of `getFloatTimeDomainData()` (or something similar)? Also wouldn't it be -1 to 1?

Reply:
Also aren't we multiplying it up again to a different thing in LiveRecordingWaveform?

Reply:
the float data was coming out super weird for me, but the byte data made sense (floats were coming out as 5 instead of -1 to 1). 

I'll test again though given there were a couple variables at the time.

Reply:
turns out I was doing Float32 to Float64 conversions wrong.